pyccel/codegen/printing/fcode.py

```diff
@@ -1787,8 +1787,20 @@ def _print_AugAssign(self, expr):
 
     def _print_PythonRange(self, expr):
         start = self._print(expr.start)
-        stop  = self._print(expr.stop) + '-' + self._print(LiteralInteger(1))
         step  = self._print(expr.step)
+
+        test_step = expr.step
+        if isinstance(test_step, PyccelUnarySub):
+            test_step = expr.step.args[0]
+
+        # testing if the step is a value or an expression
+        if isinstance(test_step, Literal):
+            stop = self._print(expr.stop) + '-' + self._print(LiteralInteger(1))
+            if isinstance(expr.step, PyccelUnarySub):
+                stop = self._print(expr.stop) + '+' + self._print(LiteralInteger(1))
+        else:
+            stop = self._print(expr.stop) + '- ( (merge('+self._print(LiteralInteger(1))+', '+self._print(LiteralInteger(-1))+', ' + self._print(expr.step) + ' > '+ self._print(LiteralInteger(0)) +')) *' + self._print(LiteralInteger(1)) + ')'
```

Comment:
this can be approved to look more readable  using PyccelNodes
```PYTHON
stop = self._print(expr.stop) + '-' + self._print(LiteralInteger(1))
```
can be 
```PYTHON
stop = PyccelAdd(expr.stop, LiteralInteger(1))
```
the same for the next condition
```PYTHON
stop = PyccelMinus(expr.stop, LiteralInteger(1))
```
and then you do 
```PYTHON
stop = self._print(stop)
```

Reply:
for Unknown  expression : 
```PTHON
            stop = IfTernaryOperator(PyccelGt(expr.step, LiteralInteger(0)), 
                PyccelMinus(expr.stop, LiteralInteger(1)),
                PyccelAdd(expr.stop, LiteralInteger(1)))
            stop = self._print(stop)

``` 
you can check operators/IfternaryOperator  docstring I let you do the same for C